project_1/part_2.py

```diff
@@ -0,0 +1,152 @@
+
+        # image 2
+        nimg2 = self.img_crop(img2)
+        aimg2 = self.img_avg(nimg2, color_channel)
+
+        # image 3
+        nimg3 = self.img_crop(img3)
+        aimg3 = self.img_avg(nimg3, color_channel)
+
+        # image 4
+        nimg4 = self.img_crop(img4)
+        aimg4 = self.img_avg(nimg4, color_channel)
+
+        # image 5
+        nimg5 = self.img_crop(img5)
+        aimg5 = self.img_avg(nimg5, color_channel)
+
+        # image 6
+        nimg6 = self.img_crop(img6)
+        aimg6 = self.img_avg(nimg6, color_channel)
+
+        # image 7
+        nimg7 = self.img_crop(img7)
+        aimg7 = self.img_avg(nimg7, color_channel)
+
+        # image 8
+        nimg8 = self.img_crop(img8)
+        aimg8 = self.img_avg(nimg8, color_channel)
+
+        # x and y axis
+        B = [aimg1, aimg2, aimg3, aimg4, aimg5, aimg6, aimg7, aimg8]
+        T = [1.0/350.0, 1.0/250.0, 1.0/180.0, 1.0/125.0, 1.0/90.0, 1.0/60.0, 1.0/45.0, 1.0/30.0]
+
+        # logs
+        log_B = np.log10(B)
+        log_T = np.log10(T)
+
+        # mean
+        mean_log_B = (np.sum(log_B))/float(len(log_B))
+        mean_log_T = (np.sum(log_T))/float(len(log_T))
+
+        return(B, T, log_B, log_T, mean_log_B, mean_log_T)
+
+    def linear_regression(self, x, y, x_bar, y_bar):
+        '''
+        Returns the linearized Y axis values
+        Args: All logarithmic values of B, T and their means
+        Return: Least squares estimate of log B, 1/g
+        '''
+        num_r = 0
+        den1_r = 0
+        den2_r = 0
+        for i in range(len(x)):
+            num_r = num_r + ((x[i] - x_bar)*(y[i] - y_bar))
+            den1_r = den1_r + ((x[i] - x_bar) ** 2)
+            den2_r = den2_r + ((y[i] - y_bar) ** 2)
+
+        den3_r = (den1_r * den2_r) ** (0.5)
+
+        # pearson's coefficient
+        r = num_r/den3_r
+
+        # standard deviation
+        sx = (den1_r/ float(len(x) - 1)) ** (0.5)
+        sy = (den2_r/ float(len(y) - 1)) ** (0.5)
+
+        # slope
+        b1 = r*(sy/sx)
+
+        # intercept
+        log_k = y_bar - b1*x_bar
+
+        return(log_k + (b1)*x, b1)
+
+    def part_1(self, color_channel):
+        B, T, log_B, log_T, mean_log_B, mean_log_T = self.part_1_init(color_channel)
+
+    # linear regression on log_B
+        linearized_log_B, g_inv = self.linear_regression(log_T, log_B, mean_log_T, mean_log_B)
+
+    # linearized B
+        linearized_B = 10**(linearized_log_B)
+
+    # plotting
+        plt.plot(log_T, linearized_log_B, label ='linearized estimate')
+        # plt.plot(log_T, log_B, label = 'observed brightness')
+        # plt.plot(T, linearized_B**(g_inv))
+        # plt.scatter(mean_log_T, mean_log_B))
+        plt.xlabel('Logarithm of Exposure Time')
+        plt.ylabel('Linearized value of logarithm of Brightness')
+        plt.savefig('log_T_vs_linearized_log_B_for_blue_channel')
+        plt.show()
+
+        return
+
+def main():
+    p = project()
+    p.part_1(1)
```

Comment:
Noice! There is also this library called args.parse which on using that we can say-

> python part_1.py green

and in the code we can set green to be 1 being sent in the argument.